src/components/TodoList.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useContext } from 'react';
+import { TodosContext } from '../store/TodosContext';
+import { TodoItem } from './TodoItem';
+import { Status } from '../types/Status';
+
+export const TodoList: React.FC = () => {
+  const { todos, filter } = useContext(TodosContext);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filter) {
+      case Status.ALL:
+        return true;
+
+      case Status.ACTIVE:
+        return !todo.completed;
+
+      case Status.COMPLETED:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
```

Comment:
No need to filter in case we should show all todos - it may cause performance issue